### PR TITLE
Properly set random seeds

### PR DESCRIPTION
**Pull request by minafarid**
* Uses `np.random.choice()` and correctly set its seed.
* Upgrade to `gensim===3.7.0` and fix corresponding API.
* Some changes for readability.

### CODE REVIEW

File: domain/domain.py [hunk 1]
```diff
@@ -138,7 +136,8 @@ def get_active_attributes(self):
         result = self.ds.engine.execute_query(query)
         if not result:
             raise Exception("No attribute contains erroneous cells.")
-        return set(itertools.chain(*result))
+        # Sort the active attributes to maintain the order of the ids of random variable.
+        return sorted(list(itertools.chain(*result)))
```
Comment (richardwu) [inline domain/domain.py]: You can omit the `list` and just have `sorted`.

File: domain/estimators/naive_bayes.py [hunk 2]
```diff
@@ -32,21 +32,21 @@ def train(self):
...
-                    val2 = row[at]
-                    val2_count = self._freq[at][val2]
-                    val2_val1_count = 0.1
-                    if val1 in self._cooccur_freq[attr][at]:
-                        if val2 in self._cooccur_freq[attr][at][val1]:
-                            val2_val1_count = max(self._cooccur_freq[attr][at][val1][val2] - 1.0, 0.1)
-                    p = float(val2_val1_count)/float(val1_count)
-                    log_prob += math.log(p)
+                # Ignore same attribute, index, and tuple id.
+                if at == attr or at == 'index' or at == '_tid_':
```
Comment (richardwu) [inline domain/estimators/naive_bayes.py]: Do we know why `index` might be included in the list of correlated attributes? We might run into issues here if there is genuinely a column in the data called `index`.

File: domain/domain.py [hunk 3]
```diff
@@ -332,52 +330,62 @@ def get_domain_cell(self, attr, row):
...
 
-        domain = set([])
+        domain = set()
         correlated_attributes = self.get_corr_attributes(attr, self.cor_strength)
         # Iterate through all attributes correlated at least self.cor_strength ('cond_attr')
         # and take the top K co-occurrence values for 'attr' with the current
         # row's 'cond_attr' value.
         for cond_attr in correlated_attributes:
+            # Ignore correlations with index, tuple id or the same attribute.
             if cond_attr == attr or cond_attr == 'index' or cond_attr == '_tid_':
```
Comment (richardwu) [inline domain/domain.py]: Do we know why `index` might be included in the list of correlated attributes? We might run into issues here if there is genuinely a column in the data called `index`.

**Review by richardwu — CHANGES_REQUESTED** [review]
Just a few minor comments 👍

File: domain/estimators/naive_bayes.py (hunk 2 above)
Comment (minafarid) [reply domain/estimators/naive_bayes.py]: it hasn't... I just noticed we do this in multiple places. I will remove the column `index` for now.

File: domain/domain.py (hunk 1 above)
Comment (minafarid) [reply domain/domain.py]: ok

File: domain/estimators/naive_bayes.py (hunk 2 above)
Comment (minafarid) [reply domain/estimators/naive_bayes.py]: I noticed we are doing this somewhere else so I followed the same convention. But I will remove it.

**Review by richardwu — APPROVED** [review]

**Pull request merged**